qutip/solver/integrator/explicit_rk.pyx

```diff
@@ -300,6 +301,7 @@ cdef class Explicit_RungeKutta:
             self._t_prev = self._t_front
             self._norm_prev = self._norm_front
             nsteps_left -= self._step_in_err(t, nsteps_left)
+            PyErr_CheckSignals()
```

Comment:
`PyErr_CheckSignals` returns `-1` if a Python signal handler raises an exception. Should we check for that and return?

Reply:
It also raise an error which is what we need.
If we check for that and return ourselves, wouldn't the error trace be lost?

Reply:
At least in C code, one would do:
```
if (PyErr_CheckSignals == -1) {
    return;
}
```
which would then return `void` and the caller would check to see if an error was set.

So my question is does Cython do this for us automatically?

Reply:
Yes cython catch the error and return with the error value without executing the rest of the function by itself.
So it should never reach the return.
Also you probably don't want to return void. With `except *`, I believe that it check for error with any return, but with with `except -1`, `-1` should be returned etc. It's better to just let cython handle that.